docs/faq/subgraphAPI.md

```diff
@@ -0,0 +1,99 @@
+
+```C++
+class SgSelector : public SubgraphSelector {
+ public:
+  SgSelector() {
+    find_bn = false;
+  }
+  bool Select(const nnvm::Node &n) override {
+    // Here we start on the convolution node to search for a subgraph.
+    return n.op() && n.op()->name == "Convolution";
+  }
+  bool SelectInput(const nnvm::Node &n, const nnvm::Node &new_node) override {
+    // We don't need to search on the incoming edge.
+    return false;
+  }
+  bool SelectOutput(const nnvm::Node &n, const nnvm::Node &new_node) override {
+    // We search on the outgoing edge. Once we find a batch_norm node, we won't
+    // accept any more batch_norm nodes.
+    if (new_node.op() && new_node.op()->name == "BatchNorm" && !find_bn) {
+      find_bn = true;
+      return true;
+    } else {
+      return false;
+    }
+  }
+  std::vector<nnvm::Node *> Filter(const std::vector<nnvm::Node *> &candidates) override {
+    // We might have found a convolution node, but we might have failed to find a batch_norm
+    // node that uses the output of the convolution node. If we failed, we should skip
+    // the convolution node as well.
+    if (find_bn)
+      return candidates;
+    else
+      return std::vector<nnvm::Node *>();
+  }
+ private:
+  bool find_bn;
+};
+```
+
+The second step is to define a subgraph property to use the subgraph selector above to customize the subgraph searching. By defining this class, we can also customize subgraph node creation. When customizing node creation, we can specify what operator to run the subgraph on the node. In this example, we use CachedOp, which itself is a graph executor, to run the subgraph with convolution and batch_norm. In practice, it's most likely that we use a single operator from a backend library to replace the two operators for execution.
+
+```C++
+class SgProperty : public SubgraphProperty {
+ public:
+  static SubgraphPropertyPtr Create() {
+    return std::make_shared<SgProperty>();
+  }
+  nnvm::NodePtr CreateSubgraphNode(
+      const nnvm::Symbol &sym, const int subgraph_id = 0) const override {
+    // We can use CachedOp to execute the subgraph.
+    nnvm::NodePtr n = nnvm::Node::Create();
+    n->attrs.op = Op::Get("_CachedOp");
+    n->attrs.name = "ConvBN" + std::to_string(subgraph_id);
+    n->attrs.subgraphs.push_back(std::make_shared<nnvm::Symbol>(sym));
+    std::vector<std::pair<std::string, std::string> > flags{{"static_alloc", "true"}};
+    n->attrs.parsed = CachedOpPtr(new CachedOp(sym, flags));
+    return n;
+  }
+  SubgraphSelectorPtr CreateSubgraphSelector() const override {
+    return std::make_shared<SgSelector>();
+  }
+};
+```
+
+After defining the subgraph property, we need to register it.
+
+```C++
+MXNET_REGISTER_SUBGRAPH_PROPERTY(SgTest, SgProperty);
+```
+
+After compiling this subgraph mechanism into MXNet, we can use the environment variable `MXNET_SUBGRAPH_BACKEND` to activate it.
+
+```bash
+export MXNET_SUBGRAPH_BACKEND=SgTest
+```
+
```

Comment:
Please add a conclusion: summary of what was covered here, relevant links, and if possible some call to action for another step or more learning...